include/reactor-uc/platform.h

```diff
@@ -29,13 +29,13 @@ struct Platform {
   lf_ret_t (*wait_for)(Platform *self, interval_t duration);
   /**
    * @brief Put the system to sleep until the wakeup time or until an
-   * asynchronous event occurs.
+   * asynchronous event occurs. Must be called from a critical section.
    */
-  lf_ret_t (*wait_until_interruptible)(Platform *self, instant_t wakeup_time);
+  lf_ret_t (*wait_until_interruptible_locked)(Platform *self, instant_t wakeup_time);
```

Comment:
does this mean before entering the function you need to lock the critical section? or does it mean this function will lock the critical section?

Reply:
`_locked` postfix means that the function only should be called when you are inside a critical section

Reply:
I guess having this stops making sense, when we add nested critical sections.

Reply:
Hm, I think there still can be functions that assume that they are within a critical section rather than acquiring it self